va/va.go

```diff
@@ -380,13 +380,15 @@ func (va *ValidationAuthorityImpl) fetchHTTP(ctx context.Context, identifier cor
 			if err != nil {
 				return err
 			}
-			if reqPort <= 0 || reqPort > 65535 {
-				return fmt.Errorf("Invalid port number %d in redirect", reqPort)
+			if reqPort != va.httpPort && reqPort != va.httpsPort {
+				return fmt.Errorf(
```

Comment:
Instead of using a normal error and using string comparison later you should use a `berrors.ConnectionFailure` here and `berrors.Is(err, berrors.ConnectionFailure)` when you want to check the type.

Reply:
SGTM